app/src/test/java/org/mozilla/focus/cfr/CfrMiddlewareTest.kt

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.focus.cfr
+
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.SecurityInfoState
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.libstate.ext.waitUntilIdle
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.doReturn
+import org.mockito.MockitoAnnotations
+import org.mozilla.focus.Components
+import org.mozilla.focus.state.AppState
+import org.mozilla.focus.state.AppStore
+import org.mozilla.focus.state.Screen
+import org.mozilla.focus.utils.AppConstants
+import org.mozilla.focus.utils.Settings
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class CfrMiddlewareTest {
+    private lateinit var cfrMiddleware: CfrMiddleware
+    private lateinit var browserStore: BrowserStore
+    private lateinit var appStore: AppStore
+    private lateinit var settings: Settings
+
+    @Mock
+    private lateinit var components: Components
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+
+        cfrMiddleware = CfrMiddleware(components)
+        browserStore = BrowserStore(
+            initialState = BrowserState(),
+            middleware = listOf(cfrMiddleware)
+        )
+        appStore = AppStore(
+            AppState(
+                screen = Screen.Home,
+                showTrackingProtectionCfr = false,
+            )
+        )
+        settings = Settings(testContext)
+
+        doReturn(appStore).`when`(components).appStore
+        doReturn(settings).`when`(components).settings
+    }
+
+    @Test
+    fun `GIVEN erase cfr is enabled and tracking protection cfr is not displayed WHEN AddTabAction is intercepted THEN the numberOfTabsOpened is increased`() {
+        if (AppConstants.isDevOrNightlyBuild) {
```

Comment:
Can you check for `IS_ERASE_CFR_ENABLED` here instead? Because then when this feature is ultimately enabled in release the tests will just run/work automatically and not require a change.